templates/routing-legacy/main.go

```diff
@@ -0,0 +1,65 @@
+// package main holds the implementation of the nextroute template.
+package main
+
+import (
+	"context"
+	"log"
+
+	"github.com/nextmv-io/sdk/nextroute"
+	"github.com/nextmv-io/sdk/nextroute/factory"
+	"github.com/nextmv-io/sdk/nextroute/schema"
+	"github.com/nextmv-io/sdk/run"
+	runSchema "github.com/nextmv-io/sdk/run/schema"
+)
+
+func main() {
+	runner := run.CLI(solver)
+	err := runner.Run(context.Background())
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+type options struct {
+	Model  factory.Options                `json:"model,omitempty"`
+	Solve  nextroute.ParallelSolveOptions `json:"solve,omitempty"`
+	Format nextroute.FormatOptions        `json:"format,omitempty"`
+}
+
+func solver(
+	ctx context.Context,
+	input schema.FleetInput,
+	options options,
+) (runSchema.Output, error) {
+	nextrouteInput, err := input.ToNextRoute()
+	if err != nil {
+		panic(err)
+	}
+
+	opt := options.Solve
+	if input.Options != nil || input.Options.Solver != nil &&
+		input.Options.Solver.Limits != nil && input.Options.Solver.Limits.Duration != nil {
```

Comment:
Shouldn't the first operator be also an and?
```suggestion
	if input.Options != nil && input.Options.Solver != nil &&
		input.Options.Solver.Limits != nil && input.Options.Solver.Limits.Duration != nil {
```

Reply:
yes